flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java

```diff
@@ -110,6 +118,42 @@ public void testFailInEndOfConstructor() throws Exception {
                                 .build(EXECUTOR_RESOURCE.getExecutor()));
     }
 
+    @Test
+    public void streamTaskAsyncExceptionHandlerHandleExceptionInInitializingDuration()
+            throws Exception {
+        MockEnvironment mockEnvironment = MockEnvironment.builder().build();
+        RuntimeException expectedException = new RuntimeException("RUNTIME EXCEPTION");
+
+        mockEnvironment.setExpectedExternalFailureCause(AsynchronousException.class);
+        final String expectedErrorMessage = "EXPECTED_ERROR MESSAGE";
+
+        NoOpStreamTask noOpStreamTask =
+                new NoOpStreamTask(mockEnvironment) {
+                    @Override
+                    protected void init() throws Exception {
+                        super.init();
+                        CompletableFuture.runAsync(
+                                () -> {
+                                    assertFalse(this.isRunning());
+                                    assertTrue(this.isInitializing());
+                                    this.handleAsyncException(
+                                            expectedErrorMessage, expectedException);
+                                });
+                    }
```

Comment:
Should the make thread of NoOpStreamTask sleep a small duration 
Or add a flag to ensure that The `runAsync` operation must run in the span (isRestoring = true, isRunning=false) ?

Reply:
Flink test doesn't recommend using the `sleep`, you can use the `CompletableFuture future = new CompletableFuture()`.

Main thread call the `future.wait()`, and async thread calls the `future.complete(null)` after some operators done.

Reply:
> Flink test doesn't recommend using the `sleep`, you can use the `CompletableFuture future = new CompletableFuture()`.

Thank you for the reminding. It graced the lines